community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v3_2/codegen/AuxGenerator.scala

```diff
@@ -75,14 +82,177 @@ class AuxGenerator(val packageName: String, val generator: CodeGenerator) {
+          val sanitizedFieldName = CodeGenContext.sanitizedName(fieldName)
+          val codeGenType = tupleDescriptor.structure(sanitizedFieldName)
+          val fieldType: TypeReference = lowerType(codeGenType)
+          val fieldReference = field(clazz.handle(), fieldType, sanitizedFieldName)
+          val thisValueName = s"thisValue_$sanitizedFieldName"
+          val otherValueName = s"otherValue_$sanitizedFieldName"
+
+          // Helper for generating code expressions for the field values to compare, with proper boxing
+          def extractFields(block: CodeBlock, reprType: RepresentationType): (Expression, Expression) = {
+            val isPrimitive = RepresentationType.isPrimitive(reprType)
+            val thisField = Expression.get(block.self(), fieldReference)
+            val otherField = Expression.get(block.load(otherTupleName), fieldReference)
+            if(isPrimitive)
+              (thisField, otherField)
+            else
+              (Expression.box(thisField), Expression.box(otherField))
+          }
+
+          // Helper for generating code for extracting the field values to compare,
+          // and assigning them to local variables
+          def assignComparatorVariablesFor(block: CodeBlock, reprType: RepresentationType): (LocalVariable, LocalVariable) = {
+            val (thisField, otherField) = extractFields(block, reprType)
+            val thisValueVariable: LocalVariable = block.declare(fieldType, thisValueName)
+            val otherValueVariable: LocalVariable = block.declare(fieldType, otherValueName)
+
+            block.assign(thisValueVariable, thisField)
+            block.assign(otherValueVariable, otherField)
+
+            (thisValueVariable, otherValueVariable)
+          }
+
+          def rearrangeInSortOrder(lhs: Expression, rhs: Expression, sortOrder: SortOrder): (Expression, Expression) =
+            sortOrder match {
+              case Ascending =>
+                (lhs, rhs)
+              case Descending =>
+                (rhs, lhs)
+            }
+
+          using(l1.block()) { l2 =>
+            codeGenType match {
+              // TODO: Primitive nodes and relationships including correct ordering of nulls
+              // TODO: Extract shared code between cases
+              case CodeGenType(symbols.CTInteger, reprType) => {
+                /*
+                E.g.
+                long thisValue_a = this.a
+                long otherValue_a = other.a
+                if (thisValue_a < otherValue_a) {
+                  return -1;
+                } else if (otherValue_a < thisValue_a) {
+                  return +1;
+                }
+                ...
+                */
+                val (thisValueVariable, otherValueVariable) = assignComparatorVariablesFor(l2, reprType)
+
+                using(l2.ifStatement(Expression.lt(thisValueVariable, otherValueVariable))) { l3 =>
+                  l3.returns(Expression.constant(lessThanSortResult(sortOrder)))
```

Comment:
`l1`, `l2`, `l3` etc. seemed to be a pre-existing convention when there were multiple nested levels of generators (`MethodStructure`). You could think of it as blocks in the generated code, and you need to emit code in the right block:
```
{ // l1
  {  // l2
    {  // l3
      // l3.returns()
      return;
    }
  }
}
```

Then again this is not followed everywhere, but at least it is _one_ convention...